### ISSUE TITLE
Rule to require `?` and matching `:` to each begin a new line if the other does

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/StyleCopAnalyzers#3330]
Please add a warning such as:

```cs
class C
{
    int M(bool cond1, bool cond2)
    {
        return cond1
            // ⚠ Matching '?' and ':' must be on the same line or each must be at the beginning of a new line
            //              ↓
            ? cond2 ? 1 : 2 : 3;
    }
}
```

They should also be required to be at the same level of indentation if they begin new lines, so this would not be valid (same rule or maybe a separate rule, maybe part of https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3331):

```cs
return cond1
    ? cond2 ? 1 : 2
        : 3;
```

The rule should not require `?` and `:` to always begin a new line because I want the option to use this pattern:

```cs
return
    cond1 ? 1 :
    cond2 ? 2 :
    cond3 ? 3 :
    4;
```

(A separate rule could, though, and I'd keep it off.)

**Comment by sharwell** [comment]
Possible duplicate of #651?

**Comment by jnm2** [comment]
Ah yes! Except for the mention of indentation, but maybe that belongs in a separate rule anyway.
